daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/primitives/PrimitivesTextNumber.scala

```diff
@@ -464,6 +464,7 @@ case class ConvertTextStandardNumberPrim(e: ElementBase)
             ),
             pattern
           )
+        case _ => Assert.impossibleCase("default case should not be reached")
```

Comment:
This kinds of `case _ => Assert.impossible` always concern me a little bit when its not clear what's impossible about it. Is scala 2.13 just not as good at detecting missing cases?

Reply:
it might not be, it was complaining about Some(_) when we had Some(something...), so I'll look into these abit more and report the errors. Is it better to address the specific cases it referenced as missing, rather than using the default case...or should I add a comment with the cases it mentioned?

Reply:
So this particular one errors out without the defailt case as below: 

match may not be exhaustive.
[error] It would fail on the following input: Some(_)
[error]       r.findFirstMatchIn(patternWithoutEscapedChars) match {